manifests/init.pp

```diff
@@ -19,240 +19,315 @@
+#   Control variable to disable promoting slave in case of disconnection from master
 #   Only set if cluster_enabled is true
-#   Default: 0
-# @param [Boolean] cluster_require_full_coverage   If false Redis Cluster will server queries even if requests about a subset of keys can be processed
+# @param cluster_require_full_coverage
+#   If false Redis Cluster will server queries even if requests about a subset of keys can be processed
 #   Only set if cluster_enabled is true
-#   Default: true
-# @param [Integer] cluster_migration_barrier    Minimum number of slaves master will remain connected with, for another slave to migrate to a  master which is no longer covered by any slave
+# @param cluster_migration_barrier
+#   Minimum number of slaves master will remain connected with, for another
+#   slave to migrate to a master which is no longer covered by any slave.
 #   Only set if cluster_enabled is true
-#   Default: 1
 class redis (
-  $activerehashing                                               = $redis::params::activerehashing,
-  $aof_load_truncated                                            = $redis::params::aof_load_truncated,
-  $aof_rewrite_incremental_fsync                                 = $redis::params::aof_rewrite_incremental_fsync,
-  $appendfilename                                                = $redis::params::appendfilename,
-  Enum['no', 'always', 'everysec'] $appendfsync                  = $redis::params::appendfsync,
-  $appendonly                                                    = $redis::params::appendonly,
-  $auto_aof_rewrite_min_size                                     = $redis::params::auto_aof_rewrite_min_size,
-  $auto_aof_rewrite_percentage                                   = $redis::params::auto_aof_rewrite_percentage,
-  Variant[Stdlib::IP::Address, Array[Stdlib::IP::Address]] $bind = $redis::params::bind,
-  $output_buffer_limit_slave                                     = $redis::params::output_buffer_limit_slave,
-  $output_buffer_limit_pubsub                                    = $redis::params::output_buffer_limit_pubsub,
-  $conf_template                                                 = $redis::params::conf_template,
+  Boolean $activerehashing                                       = true,
+  Boolean $aof_load_truncated                                    = true,
+  Boolean $aof_rewrite_incremental_fsync                         = true,
+  String $appendfilename                                         = 'appendonly.aof',
+  Enum['no', 'always', 'everysec'] $appendfsync                  = 'everysec',
+  Boolean $appendonly                                            = false,
+  String $auto_aof_rewrite_min_size                              = '64mb',
+  Integer[0] $auto_aof_rewrite_percentage                        = 100,
+  Variant[Stdlib::IP::Address, Array[Stdlib::IP::Address]] $bind = ['127.0.0.1'],
+  String $output_buffer_limit_slave                              = '256mb 64mb 60',
```

Comment:
strings should likely be `String[1]` so that you cannot pass `''`. Or `Optional[String[1]]` which would allow for a string or `undef`

Reply:
Yes, there is room for further tightening. I plan on going over them in another batch.